### ISSUE TITLE
Implement handling of sync messages from signal-cli

### ISSUE THREAD

**Issue by bezmi** [isamert/scli#26]
I am interested in working on this feature when I have time, just putting it here so it's visible. When using `signal-cli receive` from the terminal, I notice that it receives sync messages for things such as:

- other person is typing
- got/read receipts
- own messages from other devices, such as sending a message from your phone.

It would be useful to handle these, especially the last one, as it would allow seamless switching between devices, which is something I regularly do. I know the signal desktop client has this functionality, but it does not fit well with my workflow.

**Comment by isamert** [comment]
It would be great to have those features! If you need any explanation about the codebase feel free to email me (or use this issue). This is a weekend-project-turned-into-something-more-complicated, so the code may not be that easy to follow.

Also I'm not really sure that signal-cli synchronizes the messages sent from your phone. I know that it synchronizes from other clients, like Signal Desktop or another instance of signal-cli but it does not synchronize messages from the master device, which is the phone. At least it wasn't able to do in my setup, I haven't checked for a while.

**Comment by bezmi** [comment]
> Also I'm not really sure that signal-cli synchronizes the messages sent from your phone.

I have tested the latest version available in the AUR. I ran, `signal-cli -u "{NUMBER}" receive -t -1` and all of the messages from my phone since the last time I ran the same command showed up, including messages that I had sent. I don't have any other signal instances linked so it must be the phone. It also works in real-time and the sync messages pop up almost instantly.

> the code may not be that easy to follow.

I had a look at the source. The implementation for most of it seems easy enough to follow. I'm not familiar with `urwid`, so it will be something to read into when I get around to working on this.
